### PR TITLE
Fix SelectedNetworkController default metamask networkClientId

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation

Currently, the SelectedNetworkController does not set a value for the default `metamask` domain on initial `NetworkController:stateChange` event because it does not count `selectedNetworkId` as actually changed in the patch yet. This manifests itself in extension with the SelectedNetworkMiddleware setting undefined for the `req.networkClientId` attribute. ~~This PR adds one more check to the event listener to ensure that the first `NetworkController:stateChange` event is used to populate the networkClientId for the default `metamask` domain.~~ See resolution in comments

## References

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/1370

### `@metamask/selected-network-controller`

- **FIXED**: `setNetworkClientIdForDomain()` will now ignore the passed in domain value and set the networkClientId for the `metamask` domain instead when the `state.perDomainNetwork` flag is false (default)

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Comment by jiexi** [comment]
Note that this requires SelectedNetworkController to be instantiated before NetworkController. An alternative is to change `setNetworkClientIdForDomain`

from
```
  setNetworkClientIdForDomain(
    domain: Domain,
    networkClientId: NetworkClientId,
  ) {
    this.update((state) => {
      state.domains[domain] = networkClientId;
    });
  }
```

to
```
  setNetworkClientIdForDomain(
    domain: Domain,
    networkClientId: NetworkClientId,
  ) {
    this.update((state) => {
      if (state.perDomainNetwork) {
        state.domains[METAMASK_DOMAIN] = networkClientId
        return
      }
      state.domains[domain] = networkClientId;
    });
  }
```

**Comment by jiexi** [comment]
Another alternative is to instantiate SelectedNetworkController after NetworkController and then to call `selectedNetworkController.setNetworkClientIdForMetamask(networkController.state.selectedNetworkId)` afterwards

**Comment by adonesky1** [comment]
>   setNetworkClientIdForDomain(
    domain: Domain,
    networkClientId: NetworkClientId,
  ) {
    this.update((state) => {
      if (state.perDomainNetwork) {
        state.domains[METAMASK_DOMAIN] = networkClientId
        return
      }
      state.domains[domain] = networkClientId;
    });
  }
  
  Would this work? Wouldn't we be updating the the metamask domain every time then and skipping setting other domains? Did you mean `if (!state.perDomainNetwork)`? Maybe I'm missing something here?

**Review by adonesky1 — DISMISSED** [review]
LGTM

**Comment by jiexi** [comment]
>  Did you mean `if (!state.perDomainNetwork)`? Maybe I'm missing something here?

sorry, you're correct. The if statement should be negated like you suggested

**Comment by adonesky1** [comment]
Hey @jiexi sorry for the delay but
> The setNetworkClientIdForDomain approach would make it so controller instantiation order doesn't matter again

I think this  ^ approach is probably better

File: packages/selected-network-controller/tests/SelectedNetworkController.test.ts [hunk 1]
```diff
@@ -21,25 +21,26 @@ describe('SelectedNetworkController', () => {
...
-      const networkClientId = 'network1';
-      controller.setNetworkClientIdForDomain(domain, networkClientId);
-      expect(controller.state.domains[domain]).toBe(networkClientId);
+      const networkClientId = 'network2';
+      controller.setNetworkClientIdForDomain('not-metamask', networkClientId);
+      expect(controller.state.domains.metamask).toBe(networkClientId);
     });
 
-    it('can set the networkClientId for the metamask domain specifically', () => {
+    it('when the perDomainNetwork feature flag is on, it sets the networkClientId for the passed in domain', () => {
```
Comment (adonesky1) [inline packages/selected-network-controller/tests/SelectedNetworkController.test.ts:34]: [nit]
```suggestion
    it('it sets the networkClientId for the passed in domain, when perDomainNetwork state is set to true ,', () => {
```

**Review by adonesky1 — DISMISSED** [review]
Small nit otherwise LGTM

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**